code/modules/mob/living/simple_animal/hostile/thing.dm

```diff
@@ -0,0 +1,2046 @@
+	var/absorb_target = null // Thing to absorb.
+	var/points = 0 // Points to spend in upgrades. Starts with 0.
+	var/absorptions = 0 // For special abilities that require you to work hard for them.
+	var/upgrading = FALSE
+//Abilities
+	var/can_travel = TRUE // Can travel? Legacy variable.
+	var/can_weaponize = FALSE // Can weaponize?
+	var/can_convert = FALSE // Can convert those who it absorbs into monsters?
+	var/can_omnivision = FALSE // Can see everything?
+	var/can_screech = FALSE // Can screech
+	var/screech_level = 1 // Level 1 or 2
+	var/screech_cooldown = 80
+	var/can_emp = FALSE // Can EMP
+	var/emp_level = 1 // Level 1 or 2
+	var/emp_cooldown = 120
+	var/can_build = FALSE // Can build things?
+	var/creep_cooldown = 35
+	var/build_cooldown = 180 // Use a different cooldown for actual buildings
+	var/can_evolve = FALSE // Can become a big, big boy?
+
+/mob/living/simple_animal/hostile/thing/Process_Spacemove(movement_dir = 0) // Can move freely in space
+	return TRUE
+
+/mob/living/simple_animal/hostile/thing/New()
+	..()
+//	world << sound('sound/thing/warning.ogg') Make this sound play with the event that spawns it. Or play it manually.
+	var/datum/action/innate/thing/travel/travel_action = new()
+	travel_action.Grant(src)
+	var/datum/action/innate/thing/absorb/absorb_action = new()
+	absorb_action.Grant(src)
+	var/datum/action/innate/thing/empower/empower_action = new()
+	empower_action.Grant(src)
+
+/mob/living/simple_animal/hostile/thing/proc/UndefinedCommunication()
+	var/message = input(src, "Announce to your spawn", "Undefined Communication")
+	if(message)
+		for(var/mob/M in GLOB.mob_list)
+			if(istype(M, /mob/living/simple_animal/hostile/thing) || istype(M, /mob/living/simple_animal/hostile/serf) || (M in GLOB.dead_mob_list) || istype(M, /mob/living/simple_animal/hostile/bigthing))
+				to_chat(M, "<span class='shadowling'><B>Undefined Communication</b> - <B>[src]</b> says: <B>[message]</b></span>")
+
+/mob/living/simple_animal/hostile/thing/verb/Communicate()
+	set name = "Communicate"
+	set category = "Unknown"
+	set desc = "Communicate with your spawn."
+
+	UndefinedCommunication()
+
+/mob/living/simple_animal/hostile/thing/Stat()
+	..()
+	if(statpanel("Status"))
+		stat(null, "Current points: [points].")
+		stat(null, "Total absorptions: [absorptions].")
+
+/mob/living/simple_animal/hostile/thing/Life()
+	. = ..()
+	if((stat == DEAD) && (!absorbing))  // Just to be sure
+		name = "flesh"
+		desc = "Just some flesh."
+		gender = PLURAL
+	if(stat != DEAD)
+		if(!phasing)
+			name = "unknown"
+			desc = "What is that?"
+			gender = NEUTER
+		if(phasing)
+			name = "???"
+			desc = "Something is there but you can't see it."
+	phasing_cooldown--
+	sound_cooldown--
+	screech_cooldown--
+	emp_cooldown--
+	build_cooldown--
+	creep_cooldown--
+	if((can_evolve == 0) && (absorptions >= 45)) // Succesfully killed and absorbed 45 people? First of all, wow. Second of all, let's get the rest of 'em.
```

Comment:
Shouldn't this be checked everytime they get evolved instead of everytime they live? Giving them an action every life() tick and checking this condition every life() tick? Does this not sounds inefficient?